plutus-use-cases/src/Language/Plutus/Coordination/Contracts/CrowdFunding.hs

```diff
@@ -0,0 +1,150 @@
+-- | Crowdfunding contract implemented using the [[Plutus]] interface.
+-- This is the fully parallel version that collects all contributions
+-- in a single transaction. This is, of course, limited by the maximum
+-- number of inputs a transaction can have.
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE RecordWildCards   #-}
+{-# OPTIONS -fplugin=Language.Plutus.CoreToPLC.Plugin #-}
+module Language.Plutus.Coordination.Contracts.CrowdFunding (
```

Comment:
The programming model here is a bit tricky with where the validators etc show up. I wonder if we could layer something like a continuation monad on top to hide some of that? I don't really understand the domain well enough to say, though.